src/mqtt_client.rs

```diff
@@ -290,12 +281,10 @@ where
 
             let topic = topic.unwrap();
 
-            // Note: The topic may be absent at runtime (deferred `Option`).
-            let len = match self.settings.get_json(&topic, &mut data) {
-                Err(Error::Absent(_)) => continue,
-                Ok(len) => len,
-                e => e.unwrap(),
-            };
+            let payload = minimq::publication::DeferedPayload::new(|buf| {
+                // TODO: Figure out how to handle absent topics.
+                Ok(self.settings.get_json(&topic, buf).unwrap())
```

Comment:
@jordens My current thought on this is that we update the `republish_state()` iterator to allow us to simply skip absent topics. Would make this simple, so we can selectively iterate over only the present settings if we want.

Reply:
The iterator doesn't know about absent topics. We'd have to extend/change the trait.

What about just erroring from that callback and failing the packet serialization?

Reply:
I figured out how to handle this. I updated minimq to allow the error to propagate back from the `publish()` call, so if the serialization fails, we can handle it and retain the error type.